### ISSUE TITLE
[API Proposal]: DateTime.AddMonths() should have similar argument naming as other similar methods.

### ISSUE THREAD

**Issue by shaggygi** [dotnet/runtime#95222]
### Background and motivation

Method `DateTime.AddMonths(int months)` contains the word "months" where other methods include "value".

See the following lines as examples:

- **Years**: https://github.com/dotnet/runtime/blob/c23d9fa2256c11dc4ba1b1eb11c62d36b4f45159/src/libraries/System.Private.CoreLib/src/System/DateTime.cs#L1039
- **Months (issue being questioned)**: https://github.com/dotnet/runtime/blob/c23d9fa2256c11dc4ba1b1eb11c62d36b4f45159/src/libraries/System.Private.CoreLib/src/System/DateTime.cs#L980
- **Days**: https://github.com/dotnet/runtime/blob/c23d9fa2256c11dc4ba1b1eb11c62d36b4f45159/src/libraries/System.Private.CoreLib/src/System/DateTime.cs#L908
- **Minutes**: https://github.com/dotnet/runtime/blob/c23d9fa2256c11dc4ba1b1eb11c62d36b4f45159/src/libraries/System.Private.CoreLib/src/System/DateTime.cs#L961
- **Seconds**: https://github.com/dotnet/runtime/blob/c23d9fa2256c11dc4ba1b1eb11c62d36b4f45159/src/libraries/System.Private.CoreLib/src/System/DateTime.cs#L1005

This is more of a nit and most likely already discussed/decided to keep since it is a very common/stable method, but I came across it tonight and wondering why the difference.  Seems like they would be consistent following the name of property being updated or using same name like "value" in this case.

Just curious.  Thx for any insight.

### API Proposal

Basically update the method on line below to use `AddMonths(int value)` or update the other similar methods to match respective property like `AddSeconds(double seconds)`:

https://github.com/dotnet/runtime/blob/c23d9fa2256c11dc4ba1b1eb11c62d36b4f45159/src/libraries/System.Private.CoreLib/src/System/DateTime.cs#L980

### API Usage

Same usage as before just argument is named differently.

### Alternative Designs

Keep as current as no harm.

### Risks

No risk that I'm aware of.

**Comment by gfoidl** [comment]
That would be a source breaking change.

Imagine calling code like
```c#
int x = 3;
DateTime dt = dt.AddMonth(months: x);
```
then it will break after that change.

**Comment by shaggygi** [comment]
@gfoidl yup, that would be a break and wasn't thinking of that.  closing then.  thx
